experimental/dds/tree2/src/feature-libraries/editable-tree-2/editableTreeTypes.ts

```diff
@@ -280,15 +280,15 @@ export interface MapNode<TSchema extends MapSchema> extends TreeNode {
 	 * @remarks
 	 * All fields under a map implicitly exist, but `values` will yield only the fields containing one or more nodes.
 	 */
-	values(): IterableIterator<UnboxField<TSchema["mapFields"]>>;
+	values(): IterableIterator<UnboxField<TSchema["mapFields"], "notEmpty">>;
 
 	/**
 	 * Returns an iterable of key, value pairs for every entry in the map.
 	 *
 	 * @remarks
 	 * All fields under a map implicitly exist, but `entries` will yield only the entries whose fields contain one or more nodes.
 	 */
-	entries(): IterableIterator<[FieldKey, UnboxField<TSchema["mapFields"]>]>;
+	entries(): IterableIterator<[FieldKey, UnboxField<TSchema["mapFields"], "notEmpty">]>;
```

Comment:
Now that this is identical to (and redundant with) the Symbol.iterator version, do we really want both? I suspect JS maps only have both because entries predates Symbol.iterator.

Anyway, if we keep this, might be nice to add a comment in the remarks block that in most cases its more concise to use Symbol.iterator and just iterate over the map without needing to call entries.

Reply:
We want both just to match the (yes, not-entirely-thoughtfully-or-purposefully-designed) JavaScript Map API. Good idea about the comment though, I added one.